tests/test_executor.py

```diff
@@ -66,40 +87,67 @@ def test_py_dialect(self):
         self.assertEqual(generate(parse_one("x is null")), "scope[None][x] is None")
 
     def test_optimized_tpch(self):
-        for i, (sql, optimized) in enumerate(self.sqls[:20], start=1):
+        for i, (sql, optimized) in enumerate(self.tpch_sqls[:20], start=1):
             with self.subTest(f"{i}, {sql}"):
                 a = self.cached_execute(sql)
-                b = self.conn.execute(optimized).fetchdf()
+                b = self.tpch_conn.execute(optimized).fetchdf()
                 self.rename_anonymous(b, a)
                 assert_frame_equal(a, b)
 
+    def subtestHelper(self, i, table, type_exec="tpc-h"):
+        with self.subTest(f"{type_exec} {i + 1}"):
+            if type_exec == "tpc-h":
+                sql, _ = self.tpch_sqls[i]
+            else:
+                sql, _ = self.tpcds_sqls[i]
```

Comment:
Nit: maybe rewrite this a bit more compactly? Don't feel strongly about it, just gets rid of some extra visual noise.
```suggestion
            sql, _ = self.tpch_sqls[i] if tpch else self.tpcds_sqls[i]
```